File: QueryOptimizer.Models/Application/ConnectionStringModel.cs
using QueryOptimizer.Shared.Common.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace QueryOptimizer.Models.Application
{
    public class ConnectionStringModel
    {
        public DatabaseTypes Provider { get; set; }
        public string? ServerName { get; set; }
        public string? UserName { get; set; }
        public string? Password { get; set; }
        public string? DatabaseName { get; set; }
        public string? Host { get; set; }
        public int? Port { get; set; }
        public string? ServiceName { get; set; }
    }
}

File: QueryOptimizer.API/app/query-optimizer-front/src/app/models/connectionStringModel.ts
import { DatabaseTypes } from "./databaseType";

export interface ConnectionStringModel{
    provider: DatabaseTypes;
    serverName?: string | null;
    userName?: string | null;
    password?: string | null;
    databaseName?: string | null;
    host?: string | null;
    port?: number | null;
    serviceName?: string | null;
}